jsapp/js/account/plan.component.tsx

```diff
@@ -0,0 +1,440 @@
+          return interval === state.intervalFilter || price.unit_amount === 0;
+        });
+        return {
+          ...product,
+          prices: filteredPrices.length ? filteredPrices[0] : null,
+        };
+      });
+      return filterAmount.filter((product: Product) => product.prices);
+    }
+    return [];
+  };
+
+  const isSubscribedProduct = (product: Price) => {
+    if (product.prices.unit_amount === 0 && !state.subscribedProduct?.length) {
+      return true;
+    }
+    return product.name === state.subscribedProduct?.name;
+  };
+
+  const upgradePlan = (priceId: string) => {
+    if (!priceId || buttonsDisabled) {
+      return;
+    }
+    setButtonDisabled(buttonsDisabled);
+    postCheckout(priceId, state.organization?.uid)
+      .then((data) => {
+        if (!data.url) {
+          alert('There has been an issue, please try again later.');
+        } else window.location.assign(data.url);
+      })
+      .finally(() => setButtonDisabled(!buttonsDisabled));
+  };
+
+  const managePlan = () => {
+    if (!state.organization?.uid || buttonsDisabled) {
+      return;
+    }
+    setButtonDisabled(buttonsDisabled);
+    postCustomerPortal(state.organization?.uid)
+      .then((data) => {
+        if (!data.url) {
+          alert('There has been an issue, please try again later.');
+        } else window.location.assign(data.url);
+      })
+      .finally(() => setButtonDisabled(!buttonsDisabled));
+  };
+
+  // Get feature items and matching icon boolean
+  const getListItem = (listType: string, plan: string) => {
+    const listItems: {icon: boolean; item: string}[] = [];
+    filterPrices().map((price) =>
+      Object.keys(price.metadata).map((featureItem: string) => {
+        const numberItem = featureItem.lastIndexOf('_');
+        const currentResult = featureItem.substring(numberItem + 1);
+
+        const currentIcon = `feature_${listType}_check_${currentResult}`;
+        if (
+          featureItem.includes(`feature_${listType}_`) &&
+          !featureItem.includes(`feature_${listType}_check`) &&
+          price.name === plan
+        ) {
+          const keyName = `feature_${listType}_${currentResult}`;
+          let iconBool = false;
+          const itemName: string = price.metadata[keyName];
+          if (price.metadata[currentIcon] !== undefined) {
+            iconBool = JSON.parse(price.metadata[currentIcon]);
+            listItems.push({icon: iconBool, item: itemName});
+          }
+        }
+      })
+    );
+    return listItems;
+  };
+
+  const checkMetaFeatures = () => {
+    if (state.products.length > 0) {
+      filterPrices().map((price) =>
+        Object.keys(price.metadata).map((featureItem: string) => {
+          if (
+            featureItem.includes(`feature_support_`) ||
+            featureItem.includes(`feature_advanced_`) ||
+            featureItem.includes(`feature_addon_`)
+          ) {
+            setShowExpand(true);
```

Comment:
This is a computed value, there is no need to use state. The function should return a boolean instead.